File: src/jcf.challenge/Jcf.Challenge/Jcf.Challenge.Server/Models/Refueling.cs
using Jcf.Challenge.Server.Enums;
using System.ComponentModel.DataAnnotations;

namespace Jcf.Challenge.Server.Models
{
    public class Refueling : EntityBase
    {
        [Required]
        public Guid VehicleId { get; set; }
        public Vehicle Vehicle { get; set; }

        [Required]
        public Guid DriverId { get; set; }  
        public Driver Driver { get; set; }

        [Required]
        public DateTime DateRefueling { get; set; }

        [Required]
        public EFuelType FuelType { get; set; }

        [Required]
        public double Quantity { get; set; }

        [Required]
        public double PaidAmount { get; set; } = 0;

        public Refueling(Guid vehicleId, Guid driverId, DateTime dateRefueling, EFuelType fuelType, double quantity, Guid? userCreateId = null)
            : base(userCreateId)
        {
            VehicleId = vehicleId;
            DriverId = driverId;
            DateRefueling = dateRefueling;
            FuelType = fuelType;
            Quantity = quantity;
        }

        public Refueling(Guid vehicleId, Guid driverId, DateTime dateRefueling, EFuelType fuelType, double quantity, double paidAmount, Guid? userCreateId = null)
            :base (userCreateId)
        {
            VehicleId = vehicleId;            
            DriverId = driverId;
            DateRefueling = dateRefueling;
            FuelType = fuelType;
            Quantity = quantity;
            PaidAmount = paidAmount;
        }

        public void Update(Guid vehicleId, Guid driverId, DateTime dateRefueling, EFuelType fuelType, double quantity, double paidAmount, Guid? userUpdateId = null)
        {
            VehicleId = vehicleId;
            DriverId = driverId;
            DateRefueling = dateRefueling;
            FuelType = fuelType;
            Quantity = quantity;
            PaidAmount = paidAmount;

            UpdatedAt = DateTime.UtcNow;
            UserUpdateId = userUpdateId;
        }
    }
}

File: src/jcf.challenge/Jcf.Challenge/jcf.challenge.client/src/app/models/refueling.ts
import { Driver } from "./driver";
import { Vehiche } from "./vehicle";

export class Refueling {
  
  constructor(
    public id: string | null,
    public vehicleId: string,
    public vehicle: Vehiche | null,
    public driverId: string,
    public driver: Driver | null,
    public dateRefueling: Date,
    public fuelType: number,
    public quantity: number,
    public paidAmount: number
  ) {

  }

  public fuelTypeDescription(): string {
    var listFuelType =
      [
          { id: 1, name: 'Gasolina' },
          { id: 2, name: 'Etanol' },
          { id: 3, name: 'Deesel' }
      ];

    return listFuelType[this.fuelType].name;
  }
}
